devtools/maven/src/main/java/io/quarkus/maven/RunMojo.java

```diff
@@ -0,0 +1,95 @@
+    /**
+     * The list of system properties defined for the plugin.
+     */
+    @Parameter
+    Map<String, String> systemProperties = Collections.emptyMap();
+
+    @Override
+    protected boolean beforeExecute() throws MojoExecutionException, MojoFailureException {
+        return true;
+    }
+
+    @Override
+    protected void doExecute() throws MojoExecutionException, MojoFailureException {
+        Set<String> propertiesToClear = new HashSet<>();
+
+        // Add the system properties of the plugin to the system properties
+        // if and only if they are not already set.
+        for (Map.Entry<String, String> entry : systemProperties.entrySet()) {
+            String key = entry.getKey();
+            if (System.getProperty(key) == null) {
+                System.setProperty(key, entry.getValue());
+                propertiesToClear.add(key);
+            }
+        }
+
+        try (CuratedApplication curatedApplication = bootstrapApplication()) {
+            AugmentAction action = curatedApplication.createAugmentor();
+            AtomicReference<Boolean> exists = new AtomicReference<>(false);
+            AtomicReference<String> tooMany = new AtomicReference<>();
+            action.performCustomBuild(RunCommandHandler.class.getName(), new Consumer<List>() {
+                @Override
+                public void accept(List cmd) {
+                    if (cmd.get(0) instanceof NullPointerException) {
+                        exists.set(false);
+                        return;
+                    } else {
+                        exists.set(true);
+                    }
+                    if (cmd.get(0) instanceof IndexOutOfBoundsException) {
+                        tooMany.set(((IndexOutOfBoundsException) cmd.get(0)).getMessage());
+                        return;
+                    }
+                    List<String> args = (List<String>) cmd.get(0);
+                    Path workingDirectory = (Path) cmd.get(1);
+                    try {
+                        Process process = new ProcessBuilder()
+                                .command(args)
+                                .inheritIO()
+                                .directory(workingDirectory.toFile())
+                                .start();
+                        int exit = process.waitFor();
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            },
+                    RunCommandLaunchResultBuildItem.class.getName());
+            if (!exists.get()) {
+                getLog().info(
+                        "You do not need quarkus:run to execute your Quarkus project.  Just execute it like a plain Java application with java -jar");
+                return;
+            }
+            if (tooMany.get() != null) {
+                getLog().error("Run Failed: Too many installed extensions support quarkus:run.  Can only have one");
```

Comment:
should/could we have a way to explicit state which of many to use for a particular run?

Reply:
Good idea.

Reply:
Hmmm, although for lambda I wouldn't want to start the mock event server dev service.  Mind if I hold off on that until I can implement quarkus:run for Lambda?  I'd like to do that in a separate PR.